### PR TITLE
added prefix

### LINKED ISSUE #2: Mandatory key prefix
I just forgot to change the key prefix used by the `RedisMigrator` and this caused two applications to use the same migration log. That is very bad!

This leads me to the idea that the key prefix should become a mandatory part of the constructor func, forcing users to think about the potential value of this prefix.

https://github.com/fabiante/kvmigrator/blob/dcac065bbe2cec9344d6afd6bb85adb089c0b0ec/migrate.go#L16

### PR DESCRIPTION
**Pull request by krishnaduttPanchagnula**
Closes #2

### CODE REVIEW

**Comment by krishnaduttPanchagnula** [comment]
Please have alook @fabiante

File: migrate.go [hunk 1]
```diff
@@ -24,13 +24,13 @@ type RedisMigrator struct {
...
 
-func NewRedisMigrator(client *redis.Client) *RedisMigrator {
+func NewRedisMigrator(client *redis.Client, prefix string) *RedisMigrator {
 	migrator := &RedisMigrator{
 		client:     client,
 		migrations: make([]*RedisMigration, 0),
 	}
 
-	migrator.SetKeyPrefix("redis-migrator:")
+	migrator.SetKeyPrefix(prefix)
```
Comment (fabiante) [inline migrate.go:33]: Thank you for the contribution!

I haven't yet decided if we should default to `"redis-migrator:` (or better `kvmigrator:` now that this library has a public name) if users call the constructor with an empty prefix. What do you think?

Depending on the outcome that should be included in the functions documentation.

File: migrate.go [hunk 2]
```diff
@@ -24,13 +24,13 @@ type RedisMigrator struct {
 	migrationLogKey string
 }
```
Comment (fabiante) [inline migrate.go:26]: With this change I think it is time to add documentation to the function:

```suggestion

// NewRedisMigrator creates a new migrator using the given client.
//
// The given prefix is used for internal keys which the migrator maintains to manage migrations.
```

**Comment by krishnaduttPanchagnula** [comment]
Done! @fabiante . Added your suggested documentation.

**Comment by fabiante** [comment]
@krishnaduttPanchagnula I did not notice that I had to approve the github action. I just did and now it shows that the tests have to be modified according to your change.

**Comment by krishnaduttPanchagnula** [comment]
> @krishnaduttPanchagnula I did not notice that I had to approve the github action. I just did and now it shows that the tests have to be modified according to your change.

@fabiante  can you rerun the pipline , i have changed the test

**Comment by fabiante** [comment]
Nice, thank you! I will update the CHANGELOG and bump the minor version.

**Pull request merged**